src/Build/Evaluation/Evaluator.cs

```diff
@@ -2076,6 +2102,17 @@ private void EvaluateImportElement(string directoryOfImportingFile, ProjectImpor
                 PerformDepthFirstPass(importedProjectRootElement);
             }
 
+            if (!importedAnything && hasCondition)
```

Comment:
Here are some variants. You might just want to drop the  `&& hasCondition` part
```xml
  <Import Project="Foo/*"/>
  <Import Project="$(EmptyProperty)"/>
  <Import Project="Foo/*" Condition="1 == 0"/>
  <Import Project="$(EmptyProperty)" Condition="1 == 1"/>
```

---
In reply to: [118298391](https://github.com/Microsoft/msbuild/pull/2109#discussion_r118298391) [](ancestors = 118298391)